app/Services/TweetService.php

```diff
@@ -31,4 +34,19 @@ public function find($tweet_id)
     {
         return $this->tweetRep->find($tweet_id);
     }
+
+    public function search(string $searchTweet)
+    {
+        return $this->tweetRep->search($searchTweet);
+    }
+
+    public function timeline($followingTweets)
+    {
+        $tweets = $this->tweetRep->timeline($followingTweets);
+        foreach ($tweets as $tweet) {
+            $tweet->likeCount = $this->likeTweetRep->countLikes($tweet->id);
+        }
+
+        return $tweets;
```

Comment:
N+1 problem here. let's update the timeline methods to return the tweets with likes count